libtock/tock.c

```diff
@@ -317,6 +336,36 @@ allow_rw_return_t allow_readwrite(uint32_t driver, uint32_t allow, void* ptr, si
   }
 }
 
+
+allow_userspace_r_return_t allow_userspace_read(uint32_t driver,
+                                                uint32_t allow, void* ptr,
+                                                size_t size) {
+  register uint32_t r0 __asm__ ("r0")       = driver;
+  register uint32_t r1 __asm__ ("r1")       = allow;
+  register const void*    r2 __asm__ ("r2") = ptr;
+  register size_t r3 __asm__ ("r3")         = size;
+  register int rtype __asm__ ("r0");
+  register int rv1 __asm__ ("r1");
+  register int rv2 __asm__ ("r2");
+  register int rv3 __asm__ ("r3");
+  __asm__ volatile (
+    "svc 7"
+    : "=r" (rtype), "=r" (rv1), "=r" (rv2), "=r" (rv3)
+    : "r" (r0), "r" (r1), "r" (r2), "r" (r3)
+    : "memory"
+    );
+  if (rtype == TOCK_SYSCALL_SUCCESS_U32_U32) {
+    allow_userspace_r_return_t rv = {true, (void*)rv1, (size_t)rv2, 0};
+    return rv;
+  } else if (rtype == TOCK_SYSCALL_FAILURE_U32_U32) {
+    allow_userspace_r_return_t rv = {false, (void*)rv2, (size_t)rv3, (statuscode_t)rv1};
+    return rv;
+  } else {
+    // Invalid return type
+    exit(1);
```

Comment:
```suggestion
    exit(-1);
```

.. This file has a mix of `1` and `-1` for this case. I think all of these should be `-1`.., will file a separate PR for the existing ones.